lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,210 @@
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain subnets in section/group %s from External IPAM."), section_name)
+    end
+
+    # Returns an array of subnets from External IPAM. If the "Require unique subnets" setting in External IPAM
+    # is enabled, this will return one subnet(or zero if not exists). If this setting is disabled(i.e. you can
+    # have duplicate subnets and IP's), then it is possible to get more than one.
+    #
+    # Params:  1. subnet: The IPv4 or IPv6 subnet CIDR. (Examples: IPv4 - "100.10.10.0/24", IPv6 - "2001:db8:abcd:12::/124")
+    #
+    # Returns: A subnet with "data" key, or a hash with "error" otherwise
+    #
+    # Examples:
+    #   Response if subnet(s) exists:
+    #     {"code": 200, "success": true, "data": [
+    #       {"id": "32", "subnet": "10.10.10.0", "description": "Test", "mask": "29"}
+    #     ], "time": 0.007}
+    #   Response if subnet not exists:
+    #     {"error": "No subnets found"}
+    #   Response if can't connect to External IPAM server
+    #     {"error": "Unable to connect to External IPAM server"}
+    def get_subnet(subnet)
+      raise "subnet cannot be nil" if subnet.nil?
+      parse get("/get_subnet?cidr=#{subnet}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain subnet %s from External IPAM."), subnet)
+    end
+
+    # Checks whether an IP address has already been reserved in External IPAM.
+    #
+    # Inputs: 1. ip:     IP address to be checked
+    #         2. subnet: The IPv4 or IPv6 subnet CIDR. (Examples: IPv4 - "100.10.10.0/24", IPv6 - "2001:db8:abcd:12::/124")
+    #
+    # Returns: true if IP exists in External IPAM, otherwise false
+    #
+    # Proxy response Examples:
+    #   Response if IP is already reserved:
+    #     {"code": 200, "success": true, "data": [{"id":"157","subnetId":"32","ip":"10.10.10.1","is_gateway":null,"description":null,"hostname":null,"mac":null,"owner":null,"tag":"2","deviceId":null,"location":null,"port":null,"note":null,"lastSeen":null,"excludePing":"0","PTRignore":"0","PTR":"0","firewallAddressObject":null,"editDate":null,"customer_id":null}],"time":0.008}
+    #   Response if IP address is available
+    #     {"code": 200, "success": false, "message": "No addresses found", "time": 0.007}
+    #   Response if missing required parameters:
+    #     {"error": ["A 'cidr' parameter for the subnet must be provided(e.g. 100.10.10.0/24)", "Missing 'ip' parameter. An IPv4 address must be provided(e.g. 100.10.10.22)"]}
+    #   Response if subnet not exists:
+    #     {"code": 200, "success": false, "message": "No subnets found", "time": 0.007}
+    #   Response if can't connect to External IPAM server
+    #     {"error": "Unable to connect to External IPAM server"}
+    def ip_exists(ip, subnet)
+      raise "subnet cannot be nil" if subnet.nil?
+      response = parse get("/ip_exists?cidr=#{subnet}&ip=#{ip}")
```

Comment:
A more common REST way of modelling this is would be `GET /subnet/#{subnet}/#{ip}` where a 404 would indicate either the subnet or the IP doesn't exist.